src/utils/hooks/useKubevirtDataPod/useKubevirtDataPod.ts

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useMemo, useState } from 'react';
+import useWebSocket from 'react-use-websocket';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  consoleFetch,
+  K8sResourceCommon,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import useDeepCompareMemoize from '../useDeepCompareMemoize/useDeepCompareMemoize';
+
+import { constructURL, getResourceVersion, registerResourceVersion } from './utils/utils';
+import useKubevirtDataPodFilters from './useKubevirtDataPodFilters';
+
+type UseKubevirtDataPod = <T extends K8sResourceCommon>(
+  watchOptions: WatchK8sResource,
+  filterOptions?: { [key: string]: string },
+) => [T, boolean, Error];
+
+const useKubevirtDataPod: UseKubevirtDataPod = <T extends K8sResourceCommon>(
+  watchOptions: WatchK8sResource,
+  filterOptions?: { [key: string]: string },
+) => {
+  const [data, setData] = useState<T>((<unknown>[]) as T);
```

Comment:
Can you please explain to me why you used `(<unknown>[]) as T`?

Reply:
TS is quite annoying regarding default values which are empty, so example if I set a default value of [object kind of k8scommon] it will pass without need of unknown, but I dont want to give false object, so TS i forcing me to do double casting here